### ISSUE TITLE
[Internal] Integrate Danger

### ISSUE THREAD

**Issue by AliSoftware** [AliSoftware/SwiftGen#230]
At least to validate that every PR makes some changes to the `CHANGELOG.md` and check if the added entry is correctly formatted.

**Comment by AliSoftware** [comment]
Some rules not to forget when integrating Danger

* Ensure the CHANGELOG has changed when the PR contains code changes
* Ensure the CHANGELOG respect the expected format, especially a period + 2 spaces at end of line, etc
* Make Danger comment the PR to show the SwiftLint warnings if any

Note that all those should be applicable to all the repos of SwiftGen (we might share a Dangerfile for all)

**Comment by AliSoftware** [comment]
Another rule to not to forget to add when we finally take some time to add Danger:
Warn if `Podfile.lock` and `Package.resolved` are not changed together (if only one of them is changed, might be a sign that we updated dependencies in one place but forgot to update the other)

https://github.com/SwiftGen/SwiftGen/pull/411#discussion_r187089910

**Comment by AliSoftware** [comment]
This has been partially implemented in #520 but there are still some future improvements that we'd be interested in after that:

* [ ] Warn if `Podfile.lock` and `Package.resolved` are not changed together
* [ ] Make Danger comment about SwiftLint warnings
* [ ] Make Danger comment about Test results  
  (`bundle exec danger` is under the `lint` job though, so we don't have the test results in there… 🤔)
* [ ] Check the *format* of the CHANGELOG entry added (dot + 2 spaces, etc)
   _(double-check if the dedicated danger plugin can help but last time I checked the expected format it checked wasn't the same as ours)_

**Comment by djbe** [comment]
swiftlint can output to a yaml file, and danger can use that

**Comment by AliSoftware** [comment]
Yup, indeed, that's what I thought about using for item 2. (Still doesn't solve item 3 though)

Additionally, we _could_ imagine make Danger tell "everything seems good to me!" if no warning were detected, could also be a way to be welcoming to contributors…?

**Comment by djbe** [comment]
Ah, misread. We can run multiple dangers 😆

**Comment by djbe** [comment]
The "ok" message would be cool, bit of "feel good" is always nice :)

**Comment by djbe** [comment]
From #620:
- [ ] Anytime a template is changed, and not the corresponding docs --> warning.

**Comment by AliSoftware** [comment]
- [ ] Use `message()` instead of `warn` for "Big PR" message, because Danger using a :warning: sign on this kind of message could make the contributor/PR author think/wonder if they did something wrong when seeing this message, or if they need to fix something to make the PR smaller or what they need to improve… whereas it's just an informative message and there's no action required by the author for it, so the 📖 icon used by `message` would be less stressful
